mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py

```diff
@@ -0,0 +1,368 @@
+                "chroma_server_http_port"
+            ),
+        }
+
+        self._collection_name = self._connection_data.get(
+            "collection_name", "default_collection"
+        )
+        self._embedding_model_name = self._connection_data.get(
+            "embedding_function", DEFAULT_EMBEDDINGS_MODEL
+        )
+
+        self._embedding_function = load_embeddings_model(self._embedding_model_name)
+
+        self._persist_directory = self._connection_data.get("persist_directory")
+        self._collection_metadata = self._connection_data.get("collection_metadata")
+
+        self._client = chromadb.Client()
+        self.is_connected = True
+
+        VectorStoreHandler.__init__(self, name)
+
+        Chroma.__init__(
+            self,
+            client=self._client,
+            client_settings=self._connection_data,
+            embedding_function=self._embedding_function,
+            collection_name=self._collection_name,
+            collection_metadata=self._collection_metadata,
+            persist_directory=self._persist_directory,
+        )
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> API:
+        """Connect to a ChromaDB database.
+
+        Returns:
+            API: The ChromaDB _client.
+        """
+
+        if self.is_connected is True:
+            return self._client
+
+        try:
+
+            self._client = chromadb.Client(Settings(**self._client_config))
+            self.is_connected = True
+        except Exception as e:
+            log.logger.error(f"Error connecting to ChromaDB client, {e}!")
+
+        return self._client
+
+    def disconnect(self):
+        """Close the database connection."""
+
+        if self.is_connected is False:
+            return
+
+        self._client = None
+        self.is_connected = False
+
+    def check_connection(self):
+        """Check the connection to the ChromaDB database."""
+
+        responseCode = StatusResponse(False)
+        need_to_close = self.is_connected is False
+
+        try:
+            self._client.heartbeat()
+            responseCode.success = True
+        except Exception as e:
+            log.logger.error(f"Error connecting to ChromaDB , {e}!")
+            responseCode.error_message = str(e)
+        finally:
+            if responseCode.success is True and need_to_close:
+                self.disconnect()
+            if responseCode.success is False and self.is_connected is True:
+                self.is_connected = False
+
+        return responseCode
+
+    def filter_query(self, query: ASTNode) -> Dict:
+        """Converts WHERE clause to ChromaDB query syntax to filter Chromadb collection.
+
+        Args:
+            query (ASTNode): The query to filter.
+
+        Returns:
+            Dict: filter for collection.
+        """
+
+        where = {}
+        if query.where.op == "and":
```

Comment:
I'm having a bit of trouble reading this 😅 . Would it work for an `AND` chain? e.g. `WHERE x=x1 AND y=y1 AND z=z1`? Not a must have, just curious (and would be good to document in the handler's `README.md` perhaps).